lib/v2/yicai/dt.js

```diff
@@ -0,0 +1,107 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+const types = {
+    article: 2,
+    report: 3,
+    visualization: 4,
+};
+
+module.exports = async (ctx) => {
+    const { type = 'article', category = '0' } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 30;
+
+    const rootUrl = 'https://dt.yicai.com';
+    const apiUrl = new URL('api/getNewsList', rootUrl).href;
+    const currentUrl = new URL(type, rootUrl).href;
+
+    const { data: response } = await got(apiUrl, {
+        searchParams: {
+            page: 1,
+            rid: types[type],
+            cid: category,
+            pageSize: limit,
+        },
+    });
+
+    let items = response.data.data.slice(0, limit).map((item) => ({
+        title: item.newstitle,
+        link: new URL(item.url, rootUrl).href,
+        description: art(path.join(__dirname, 'templates/description.art'), {
+            image: {
+                src: item.originPic,
+                alt: item.newstitle,
+            },
+            intro: item.newsnotes,
+        }),
+        author: item.creatername,
+        category: [item.channelrootname, item.channelname, item.NewsTypeName].filter((c) => c),
+        guid: `yicai-dt-${item.newsid}`,
+        pubDate: parseDate(item.utc_createdate),
+        updated: parseDate(item.utc_lastdate),
+        enclosure_url: item.originVideo,
+        enclosure_type: item.originVideo ? `video/${item.originVideo.split(/\./).pop()}` : undefined,
+        upvotes: item.newsscore ?? 0,
+    }));
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: detailResponse } = await got(item.link);
+
+                const content = cheerio.load(detailResponse);
+
+                content('div.logintips').remove();
+
+                content('img').each((_, e) => {
+                    e = content(e);
+
+                    content(e).replaceWith(
+                        art(path.join(__dirname, 'templates/description.art'), {
+                            image: {
+                                src: e.prop('data-original') ?? e.prop('src'),
+                                alt: e.prop('alt'),
+                                width: e.prop('width'),
+                                height: e.prop('height'),
+                            },
+                        })
+                    );
+                });
+
+                item.description += art(path.join(__dirname, 'templates/description.art'), {
+                    description: content('div.txt').html(),
+                });
+                item.author = content('div.authortime h3').text();
+                item.enclosure_url = '';
+                item.enclosure_type = 'video/mp4';
```

Comment:
The above value got overwritten before use.
https://github.com/DIYgod/RSSHub/blob/73867c3ffcaab672f292ff95c3e2dea02bd48c3d/lib/v2/yicai/dt.js#L45-L46